tests/phpunit/tests/block-template.php

```diff
@@ -11,10 +11,48 @@
  * @group block-templates
  */
 class Tests_Block_Template extends WP_UnitTestCase {
-	private static $post;
+	private static $post_id;
 
 	private static $template_canvas_path = ABSPATH . WPINC . '/template-canvas.php';
 
+	private static $demo_block_template;
+	private static $demo_post_content;
+	private static $demo_parsed_template;
+	private static $demo_parsed_content;
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$demo_block_template  = '<!-- wp:group {"tagName":"main"} --> <main class="wp-block-group">';
+		self::$demo_block_template .= '<!-- wp:post-title {"level":1,"align":"wide"} /-->';
+		self::$demo_block_template .= '<!-- wp:spacer {"height":32} --> <div style="height:32px" aria-hidden="true" class="wp-block-spacer"></div> <!-- /wp:spacer -->';
+		self::$demo_block_template .= '<!-- wp:post-content {"layout":{"inherit":true}} /-->';
+		self::$demo_block_template .= '</main> <!-- /wp:group -->';
+
+		self::$demo_post_content  = '<!-- wp:image {"width":1500,"height":1500,"sizeSlug":"large"} -->';
+		self::$demo_post_content .= '<figure class="wp-block-image size-large"><img src="https://picsum.photos/1500" alt="" width="1500" height="1500"/></figure>';
+		self::$demo_post_content .= '<!-- /wp:image -->';
+
+		self::$post_id = $factory->post->create(
+			array(
+				'post_title'   => 'Demo Block Editor Post',
+				'post_content' => self::$demo_post_content,
+				'post_type'    => 'post',
+				'post_status'  => 'publish',
+			)
+		);
+
+		self::$demo_parsed_content = '<figure class="wp-block-image size-large"><img decoding="async" src="https://picsum.photos/1500" alt="" width="1500" height="1500" /></figure>';
+
+		self::$demo_parsed_template  = ' <main class="is-layout-flow wp-block-group">';
+		self::$demo_parsed_template .= '<h1 class="alignwide wp-block-post-title">Demo Block Editor Post</h1>';
+		self::$demo_parsed_template .= ' <div style="height:32px" aria-hidden="true" class="wp-block-spacer"></div> ';
+		self::$demo_parsed_template .= '<div class="is-layout-constrained entry-content wp-block-post-content">' . self::$demo_parsed_content . '</div>';
+		self::$demo_parsed_template .= '</main> ';
+	}
+
+	public static function wpTearDownAfterClass() {
+		wp_delete_post( self::$post_id, true );
```

Comment:
I thought the parent method deletes all posts. Curious, do other tests in the test suite fail without this line?